Felienne/hedy
# Add type behaviour for != operator #1407

**Description**

- Refactor hash_var to not contain accidental dead code
- Removed duplicated code from transpiler (check_print_arguments)
- Allow != to have work with all types: text, number, list
- Give error if != is comparing diff types, e.g. number and text. The only automatic type coercion is int -> float.

**Fix for**

#1407 

**How to test**

- Tests are added to cover the behaviour of !=
- Open Hedy level 14 and run the following code. Double-check that the message provides enough info.
```
a is 'test'
b is 5
if a != b
   b is 1
```

**Checklist**

If you're unsure about any of these, don't hesitate to ask. We're here to help!
